src/node.h

```diff
@@ -491,6 +491,21 @@ struct IsolateSettings {
       allow_wasm_code_generation_callback = nullptr;
   v8::ModifyCodeGenerationFromStringsCallback2
       modify_code_generation_from_strings_callback = nullptr;
+
+  // When the settings is passed to NewIsolate():
+  // - If cpp_heap is not nullptr, this CppHeap will be used to create
+  //   the isolate and its ownership will be passed to V8.
+  // - If this is nullptr, Node.js will create a CppHeap that will be
+  //   owned by V8.
+  //
+  // When the settings is passed to SetIsolateUpForNode():
+  // cpp_heap will be ignored. Embedders must ensure that the
+  // v8::Isolate has a CppHeap attached while it's still used by
+  // Node.js, for example using v8::CreateParams.
+  //
+  // See https://issues.chromium.org/issues/42203693. In future version
+  // of V8, this CppHeap will be created by V8 if not provided.
+  v8::CppHeap* cpp_heap = nullptr;
```

Comment:
This needs a flag in `IsolateSettingsFlags` (similar to e.g. `ALLOW_MODIFY_CODE_GENERATION_FROM_STRINGS_CALLBACK`), as otherwise it's ABI-breaking? Or do you want to do that when backporting this change?

Reply:
I think the new tracing mechanism is already ABI breaking, so this can't be backported anyway?

Reply:
Also this is part of the migration that builds on top of the new `v8::Object::Wrap`/`v8::Object::Unwrap` APIs, which won't be available until 12.5 with https://issues.chromium.org/issues/328117814 (I suspect the series of changes are already ABI-breaking in themselves somehow....or it would be very difficult to backport all the related patches in a non-ABI breaking way to older releases).